Tasks: Add image-text-to-text pipeline and inference API to task page (#1039)
huggingface/huggingface.js: packages/tasks/src/tasks/image-text-to-text/about.md

## Inference

> You can use the Transformers library to interact with vision-language models. Specifically `pipeline` makes it easy to infer models.

Reviewer: ```suggestion
You can use the Transformers library to interact with vision-language models. Specifically, `pipeline` makes it easy to infer models.
```

> You can use the Transformers library to interact with vision-language models. Specifically, `pipeline` makes it easy to infer models.

Reviewer 2: ```suggestion
You can use the Transformers library to interact with [vision-language models](https://huggingface.co/models?pipeline_tag=image-text-to-text&transformers). Specifically, `pipeline` makes it easy to infer models.
```

Suggestion to link the models people can run via the snipper below.

Initialize the pipeline first.

```python
from transformers import pipeline

> pipe = pipeline("image-text-to-text", model="llava-hf/llava-interleave-qwen-0.5b-hf")

Reviewer: any suggestions for a good example cc @Vaibhavs10? (I'm fine with this small one)

Author: we are trying to keep the barrier of entry low with this pages per compute. we could do small internVL2.5 otherwise.

```

> We will use chat templates to format the text input. We can also pass the image as URL in context part of the user role in our chat template.

Reviewer: ```suggestion
The model's built-in chat template will be used to format the conversational input. We can pass the image as an URL in the `content` part of the user message:
```

```python
messages = [
     {
         "role": "user",
         "content": [
             {
                 "type": "image",
                 "image": "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/bee.jpg",
             },
             {"type": "text", "text": "Describe this image."},
         ],
     },
     {
         "role": "assistant",
         "content": [
             {"type": "text", "text": "There's a pink flower"},
         ],

>      },

Reviewer: It's a bit strange to me that the input ends with an assistant turn. I see in the example later that the model completes the sentence with more details, but I'm not sure this is compatible with all chat VLMs. Can we maybe skip the `assistant` role from the input and see if the model provides a good description of the image?

Reviewer: This has not been addressed, I think it's unusual that users supply an assistant turn with the input.

Author: sorry I thought I answered to this. basically it's to give more control to further align the output during inference. I used the same example here where you can see the output https://huggingface.co/docs/transformers/en/tasks/image_text_to_text

Author: sorry I think I should've sent the particular title, here you go https://huggingface.co/docs/transformers/en/tasks/image_text_to_text#pipeline I meant this one

 ]

```

> We can now directly pass in the messages to pipeline to infer. `return_full_text` is a flag to include the full prompt including the user input. Here we pass as `False` to only return the generated part.

Reviewer: ```suggestion
We can now directly pass in the messages to the pipeline to infer. The `return_full_text` flag is used to return the full prompt in the response, including the user input. Here we pass `False` to only return the generated text.
```

```python
outputs = pipe(text=messages, max_new_tokens=20, return_full_text=False)

outputs[0]["generated_text"]
# with a yellow center in the foreground. The flower is surrounded by red and white flowers with green stems
```

> You can also use Inference API to play with image-text-to-text models.

Reviewer: ```suggestion
You can also use the Inference API to test image-text-to-text models. You need to use a [Hugging Face token](https://huggingface.co/settings/tokens) for authentication.
```
